tsdb/store.go

```diff
@@ -957,6 +946,20 @@ func IsRetryable(err error) bool {
 	return true
 }
 
+// DecodeStorePath extracts the database and retention policy names
+// from a given shard or WAL path.
+func DecodeStorePath(shardOrWALPath string) (database, retentionPolicy string) {
+	// shardOrWALPath format: /maybe/absolute/base/then/:database/:retentionPolicy/:nameOfShardOrWAL
+
+	// Discard the last part of the path (the shard name or the wal name).
+	path, _ := filepath.Split(filepath.Clean(shardOrWALPath))
+
+	// Extract the database and retention policy.
+	path, rp := filepath.Split(filepath.Clean(path))
```

Comment:
It is necessary.  See https://github.com/golang/go/issues/9928 for a slightly more detailed discussion.

`filepath.SplitList` is meant to parse out colon separated list of paths like you would see in your `$PATH` environment variable.